src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -73,9 +69,30 @@ ScheduleMap GcsPackStrategy::Schedule(
   return schedule_map;
 }
 
-/// This is an initial algorithm to respect spread algorithm.
-/// In this algorithm, we try to spread all the bundle in different node
-/// and don't care the real resource.
+ScheduleMap GcsPackStrategy::Schedule(
+    std::vector<std::shared_ptr<ray::BundleSpecification>> &bundles,
+    const std::unique_ptr<ScheduleContext> &context) {
+  // The current algorithm is to select a node and deploy as many bundles as possible.
+  // First fill up a node. If the node resource is insufficient, select a new node.
+  ScheduleMap schedule_map;
+  const auto &alive_nodes = context->node_manager_.GetClusterRealtimeResources();
+  for (const auto &bundle : bundles) {
+    const auto &required_resources = bundle->GetRequiredResources();
+    for (auto &node : alive_nodes) {
```

Comment:
Consider adding a TODO to speed this up (currently it a double for loop).

Reply:
fixed